src/server/server.js

```diff
@@ -34,6 +35,8 @@ const app = express();
  * fix. */
 global.atob = atob;
 
+app.set('query parser', str => qs.parse(str, { arrayLimit: 1000 }));
```

Comment:
@elfman It looks like you found the correct reason behind the issue, but I don't like this way solving it. The reason `qs` limits possible array lengths and turns array elements with indices larger than 20 into objects is the following: say you set limit 1000, then it is legit to send query param x[999]=... and the backend will have to create an array with 1000 elements for this param. I guess, it opens a very easy way to spam the server, if somebody wants to: you just send in a query like x[999]=a&y[999]=b&... and server will have lots of fun creating a big array for each of params.
So, I guess, we better just account in the reducer that long arrays are actually parsed as objects.